covfee/server/orm/task.py

```diff
@@ -138,18 +138,25 @@ def make_results_dict(self):
         for response in self.responses:
             result_dict = response.make_results_dict()
 
-            result_dict["annotations"] = [
-                utils.NoIndentJSON(annotation.data_json)
-                for annotation in self.annotations
-                if annotation.data_json is not None
-            ]
+            # FIXME: #CONFLAB: do this loop for the getattr in the annotations class
+            result_dict["annotations"] = {}
+            for annotation in self.annotations:
+                if annotation.data_json is not None:
+                    result_dict["annotations"][annotation.id] = {
+                        "participant": annotation.participant,
+                        "category": annotation.category,
+                        "data": utils.NoIndentJSON(annotation.data_json),
+                    }
 
             prolific_ids = []
+            journeys_global_unique_id = []
             for journey in self.journeys:
                 annotator = journey.annotator
                 if annotator is not None and annotator.prolific_id is not None:
```

Comment:
in the "else" case, I would either still append an empty string to "prolific_ids" so there is a 1:1 match between the prolific_ids list and the journeys_global_unique_id list. Or make a "journeys" entry into the results_dict, such that you have results["journeys"]["prolific_id"] and results["journeys"]["global_unique_id"]

Reply:
d6cd36050337d145e1f3b52490eb4e3b450df8d3